### ISSUE TITLE
Pytorch compile failure on Windows with CUDA 12.1 because of lacking NVTX component

### ISSUE THREAD

**Issue by ain-soph** [pytorch/pytorch#101135]
### 🐛 Describe the bug

**Procedure to reproduce:**
1. For a new windows system
2. Install Miniconda and create a python=3.11 environment
3. Install cuda 12.1 with all components
4. git clone pytorch and run `python setup.py develop`
5. It fails.

**Reason:** There is no longer NVTX component in cuda 12.1, while pytorch assumes it installed at `C:\Program Files\NVIDIA Corporation\NvToolsExt`.

**Solution:** Open CUDA 11.8 installer and install NVTX component.

**Expected Behavior:** Pytorch shall be compiled with only CUDA 12.1. We don't need to install an extra NVTX component in CUDA 11.8.

**Error Log:**
```
Building wheel torch-2.1.0a0+git4052741
-- Building version 2.1.0a0+git4052741
cmake -GNinja -DBUILD_PYTHON=True -DBUILD_TEST=True -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=D:\workspace\pytorch\torch -DCMAKE_PREFIX_PATH=C:\Users\ain-s\miniconda3\Lib\site-packages -DJAVA_HOME=C:\Program Files\Java\jdk1.8.0_301 -DNUMPY_INCLUDE_DIR=C:\Users\ain-s\miniconda3\lib\site-packages\numpy\core\include -DPYTHON_EXECUTABLE=C:\Users\ain-s\miniconda3\python.exe -DPYTHON_INCLUDE_DIR=C:\Users\ain-s\miniconda3\Include -DPYTHON_LIBRARY=C:\Users\ain-s\miniconda3/libs/python310.lib -DTORCH_BUILD_VERSION=2.1.0a0+git4052741 -DUSE_NUMPY=True D:\workspace\pytorch
CMake Warning at CMakeLists.txt:368 (message):
  TensorPipe cannot be used on Windows.  Set it to OFF


-- Current compiler supports avx2 extension. Will build perfkernels.
-- Current compiler supports avx512f extension. Will build fbgemm.
CMake Error at cmake/public/cuda.cmake:69 (message):
  Failed to find nvToolsExt
Call Stack (most recent call first):
  cmake/Dependencies.cmake:43 (include)
  CMakeLists.txt:710 (include)


-- Configuring incomplete, errors occurred!
```

### Versions

I've installed successfully. If you do need the script output in error case, I could uninstall NVTX and try it days later.

cc @malfet @seemethere @peterjc123 @mszhanyi @skyline75489 @nbcsm @vladimir-aubrecht

**Comment by andreigh** [comment]
I have just hit this issue trying to compile PyTorch on Windows with Cuda 12.1. I am thinking fixing this by changing the existing references to NVTX to the one already included in nccl ? ([src/include/nvtx3](https://github.com/NVIDIA/nccl/tree/9b7d5edbfcd70a1ab0874e27382d5c31f769aa0f/src/include/nvtx3)) Does this sound reasonable ?
Alternatively, I can add a third_party reference to https://github.com/NVIDIA/NVTX, and change all references to that.

**Comment by andreigh** [comment]
Using this [patch](https://gist.github.com/andreigh/f78f631e0039f0af410b269acbb0c8dc), it compiles successfully. I'll create a PR, too.

**Comment by andreigh** [comment]
@ain-soph can you please take a look at #106763 ? Thank you !

**Comment by andreigh** [comment]
@peterbell10 I see you've done quite a bit of CMakeFiles.txt changes, is there any chance you can take a quick look at #106763 please ? Thank you

**Comment by peterbell10** [comment]
Looks like #97582 would fix this

**Comment by ke5fx** [comment]
This still seems to be broken as of December 28, four months after the original report...

**Comment by pbelcak** [comment]
@peterbell10 @mikaylagawarecki I second @ke5fx ; this is till an issue trying to build pytorch on Windows as of Jan 6th.

**Comment by mantaionut** [comment]
Tested this and it is fixed now that #109843 landed.
